src/cript/api/api.py

```diff
@@ -0,0 +1,234 @@
+from cript.nodes.supporting_nodes.group import Group
+from cript.nodes.supporting_nodes.user import User
+
+_global_cached_api = None
+
+
+def _get_global_cached_api():
+    """
+    Read-Only access to the globally cached API object.
+    Raises an exception if no global API object is cached yet.
+    """
+    if _global_cached_api is None:
+        raise CRIPTAPIAccessError
+    return copy.copy(_global_cached_api)
+
+
+class API:
+    _host: str = ""
+    _token: str = ""
+
+    def __init__(self, host: Union[str, None], token: [str, None]) -> None:
+        """
+        Initialize object with host and token.
+        It is necessary to use a `with` context manager with the API like so:
+        ```
+        with cript.API('https://criptapp.org', 'secret_token') as api:
+           # node creation, api.save(), etc.
+        ```
+
+        Parameters
+        ----------
+        host : str, None
+            CRIPT host to connect to such as "https://criptapp.org"
+            if host ends with a "/" such as "https://criptapp.org/"
+            then it strips it to always be uniform.
+            This host address is the same that use to login to cript website.
+
+            If `None` is specified, the host is inferred from the environment variable `CRIPT_HOST`.
+
+        token : str, None
+            CRIPT API Token used to connect to CRIPT
+            You can find your personal token on the cript website at User > Security Settings.
+            The user icon is in the top right.
+            If `None` is specified, the token is inferred from the environment variable `CRIPT_TOKEN`.
+
+
+        Warns
+        -----
+        UserWarning
+            If `host` is using "http".
+
+        Raises
+        ------
+        ConnectionError
+            If it cannot connect to CRIPT with the provided host and token
+
+        Returns
+        -------
+        None
+        """
+
+        if host is None:
+            host = os.environ.get("CRIPT_HOST")
+            if host is None:
+                raise RuntimeError(
+                    "API initilized with `host=None` but environment variable `CRIPT_HOST` not found.\n"
+                    "Set the environment variable (preferred) or specify the host explictly at the creation of API."
+                )
+        if token is None:
+            token = os.environ.get("CRIPT_TOKEN")
+            if token is None:
+                raise RuntimeError(
+                    "API initilized with `token=None` but environment variable `CRIPT_TOKEN` not found.\n"
+                    "Set the environment variable (preferred) or specify the token explictly at the creation of API."
+                )
+        # strip ending slash to make host always uniform
+        host = host.rstrip("/")
+
+        # if host is using unsafe "http://" then give a warning
+        if host.startswith("http://"):
+            warnings.warn("HTTP is an unsafe protocol please consider using HTTPS.")
```

Comment:
I would recommend using pythons logging module; its a lot more flexible and better as things get more complex. Its also nice for people who are going to build upon this package for getting log reports for debugging.

Reply:
There is actually a page that lists, how to log errors in python and which module to use.
https://docs.python.org/3/howto/logging.html

To me it seems like that `warnings.warn` is the right decision here.
The "client application" can do something about this: The user script can specify https instead of http.
So warnings is appropriate.